core/pfe-core/controllers/internals-controller.ts

```diff
@@ -0,0 +1,21 @@
+import type { ReactiveController, ReactiveControllerHost } from 'lit';
+
+export class InternalsController implements ReactiveController {
+  #internals: ElementInternals;
+
+  constructor(
+    public host: ReactiveControllerHost & HTMLElement,
+  ) {
+    this.#internals = host.attachInternals();
+  }
+
+  hostConnected?(): void
+
+  submit() {
+    this.#internals.form?.requestSubmit(this.host);
+  }
+
+  reset() {
+    this.#internals.form?.reset();
+  }
```

Comment:
🔥 almost forgot that `<input>` also has submit and reset while reviewing, at first was thinking these should be in a subclass of `InternalsControlller` like `ButtonInternals`.  I think this abstraction will be nice to place common controls in.